[Queues] Update docs with `contentType` option in "platform/javascript-apis.md" (#9975)
cloudflare/cloudflare-docs: content/queues/platform/javascript-apis.md

  - The body of the message.
  - The body can be any type supported by the [structured clone algorithm](https://developer.mozilla.org/en-US/docs/Web/API/Web_Workers_API/Structured_clone_algorithm#supported_types), as long as its size is less than 128 KB.

- {{<code>}}contentType{{<param-type>}}QueuesContentType{{</param-type>}}{{</code>}}

  - The explicit content type of a message so it can be previewed correctly with the [List messages from the dashboard](/queues/examples/list-messages-from-dash/) feature. Optional argument.
  - As of now, this option is for internal use. In the future, `contentType` will be used by alternative consumer types to explicitly mark messages as serialized so they can be consumed in the desired type.
  - See [QueuesContentType](#QueuesContentType) for possible values.

{{</definitions>}}

### `QueuesContentType`

A union type containing valid message content types.

```ts

> type QueuesContentType = "text" | "bytes" | "json" | "v8";

Reviewer: We should explain when each of these is relevant, and what happens if you select the wrong one.

type QueuesContentType = "text" | "bytes" | "json" | "v8";
```

> - Use `"text"` if you're sending a string. This content type can be previewed with the [List messages from the dashboard](/queues/examples/list-messages-from-dash/) feature.

Reviewer 2: ```suggestion
- Use `"text"` to send a `String`. This content type can be previewed with the [List messages from the dashboard](/queues/examples/list-messages-from-dash/) feature.
```

> - Use `"json"` if you're sending a JavaScript object that can be JSON-serialized. This content type can be previewed from the dashboard.

Reviewer 2: ```suggestion
- Use `"json"` to send a JavaScript object that can be JSON-serialized. This content type can be previewed from the [Cloudflare dashboard](https://dash.cloudflare.com).
```

> - Use `"bytes"` if you're sending an `ArrayBuffer`. This content type cannot be previewed from the dashboard and will display as Base64-encoded.

Reviewer 2: ```suggestion
- Use `"bytes"` to send an `ArrayBuffer`. This content type cannot be previewed from the [Cloudflare dashboard](https://dash.cloudflare.com) and will display as Base64-encoded.
```

> - Use `"v8"` if you're sending a JavaScript object that cannot be JSON-serialized but is supported by [structured clone](https://developer.mozilla.org/en-US/docs/Web/API/Web_Workers_API/Structured_clone_algorithm#supported_types) (e.g. `Date`s, `Map`s). This content type cannot be previewed from the dashboard and will display as Base64-encoded. `"v8"` is the default content type.

Reviewer 2: ```suggestion
- Use `"v8"` to send a JavaScript object that cannot be JSON-serialized but is supported by [structured clone](https://developer.mozilla.org/en-US/docs/Web/API/Web_Workers_API/Structured_clone_algorithm#supported_types) (for example `Date`and `Map`). This content type cannot be previewed from the [Cloudflare dashboard](https://dash.cloudflare.com) and will display as Base64-encoded. `"v8"` is the default content type.
```
